### DISCUSSION TITLE
About conan 2.0 and related CI issues

### DISCUSSION THREAD

**Discussion by matrixji** [milvus-io/milvus#22489] [Q&A and General discussion] [answered]
As [Conan](https://pypi.org/project/conan/) 2.0.0 is released this Feb.

The conan 2.0 has some broken changes, I have some trouble with compiling Milvus from the source.

By now we using pip to install the latest conan (`pip3 install conan`), Currently, I simply downgrade canon to "<2.0" to resolve the compiling problems.
And another question is why CI still works fine? I guess CI uses some prebuilt docker images as the build env, so it keeps using the conan 1.

So my questions are:
- How we go next, still using conan 1 or upgrade to conan 2.
- How does CI detect such broken changes, do we need some basic principles to ensure keep every community developer has the same or similar development env, that could save our lives

**Comment by jiaoew1991** [comment, marked as answer]
we have locked the version of Conan == 1.58.0 in the builder images, so the CI is not affected.
I will migrate conan1 to conan2 when all packages we depend on are ready.

**Reply by matrixji** [reply]
Thanks for the good answer to Question 1.
For the 2nd point, that is to say, we should keep `conan==1.58.0` in the scripts like `install_deps.sh` instead of the current: `pip3 install conan`

**Reply by jiaoew1991** [reply]
I opened a pr for that https://github.com/milvus-io/milvus/pull/22503 😄
